eras/shelley/impl/testlib/Test/Cardano/Ledger/Shelley/ImpTest.hs

```diff
@@ -934,14 +1086,24 @@ fixupFees tx = impAnn "fixupFees" $ do
             & bodyTxL . feeTxBodyL .~ (fee <> change)
   pure txWithFee
 
+-- | Adds an auxiliary data hash if auxiliary data present, while and the hash of it is not.
+fixupAuxDataHash :: (EraTx era, Applicative m) => Tx era -> m (Tx era)
+fixupAuxDataHash tx
+  | SNothing <- tx ^. bodyTxL . auxDataHashTxBodyL
+  , SJust auxData <- tx ^. auxDataTxL =
+      pure (tx & bodyTxL . auxDataHashTxBodyL .~ SJust (AuxiliaryDataHash (hashAnnotated auxData)))
+  | otherwise = pure tx
+
 shelleyFixupTx ::
   forall era.
   (ShelleyEraImp era, HasCallStack) =>
   Tx era ->
   ImpTestM era (Tx era)
 shelleyFixupTx =
   addNativeScriptTxWits
+    >=> fixupAuxDataHash
```

Comment:
That's right. Instead of manually supplying the hash for auxiliary data, I decided to create a general fixup function that will take care of it automatically